apis/kubedb/v1alpha2/mongodb_helpers.go

```diff
@@ -438,44 +438,64 @@ func (m *MongoDB) SetTLSDefaults() {
 		return
 	}
 
+	defaultServerOrg := []string{KubeDBOrganization}
 	if m.Spec.ShardTopology != nil {
+		_, cert := kmapi.GetCertificate(m.Spec.TLS.Certificates, string(MongoDBServerCert))
+		if cert != nil && cert.Subject != nil && cert.Subject.Organizations != nil {
+			defaultServerOrg = []string{}
+		}
+
 		m.Spec.TLS.Certificates = kmapi.SetMissingSpecForCertificate(m.Spec.TLS.Certificates, kmapi.CertificateSpec{
 			Alias:      string(MongoDBServerCert),
 			SecretName: "",
 			Subject: &kmapi.X509Subject{
-				Organizations:       []string{KubeDBOrganization},
-				OrganizationalUnits: []string{string(MongoDBServerCert)},
+				Organizations: defaultServerOrg,
 			},
 		})
+
 		// reset secret name to empty string, since multiple secrets will be created for each StatefulSet.
 		m.Spec.TLS.Certificates = kmapi.SetSecretNameForCertificate(m.Spec.TLS.Certificates, string(MongoDBServerCert), "")
 	} else {
+		_, cert := kmapi.GetCertificate(m.Spec.TLS.Certificates, string(MongoDBServerCert))
+		if cert != nil && cert.Subject != nil && cert.Subject.Organizations != nil {
+			defaultServerOrg = []string{}
+		}
 		m.Spec.TLS.Certificates = kmapi.SetMissingSpecForCertificate(m.Spec.TLS.Certificates, kmapi.CertificateSpec{
 			Alias:      string(MongoDBServerCert),
 			SecretName: m.CertificateName(MongoDBServerCert, ""),
 			Subject: &kmapi.X509Subject{
-				Organizations:       []string{KubeDBOrganization},
-				OrganizationalUnits: []string{string(MongoDBServerCert)},
+				Organizations: defaultServerOrg,
 			},
 		})
 	}
+
+	defaultClientOrg := []string{KubeDBOrganization}
+	_, cert := kmapi.GetCertificate(m.Spec.TLS.Certificates, string(MongoDBClientCert))
+	if cert != nil && cert.Subject != nil && cert.Subject.Organizations != nil {
+		defaultClientOrg = []string{}
+	}
 	m.Spec.TLS.Certificates = kmapi.SetMissingSpecForCertificate(m.Spec.TLS.Certificates, kmapi.CertificateSpec{
 		Alias:      string(MongoDBClientCert),
 		SecretName: m.CertificateName(MongoDBClientCert, ""),
 		Subject: &kmapi.X509Subject{
-			Organizations:       []string{KubeDBOrganization},
-			OrganizationalUnits: []string{string(MongoDBClientCert)},
+			Organizations: defaultClientOrg,
 		},
 	})
+
+	defaultExporterOrg := []string{KubeDBOrganization}
+	_, cert = kmapi.GetCertificate(m.Spec.TLS.Certificates, string(MongoDBMetricsExporterCert))
+	if cert != nil && cert.Subject != nil && cert.Subject.Organizations != nil {
+		defaultExporterOrg = []string{}
```

Comment:
Why not just assign `cert.Subject.Organizations` ?

Reply:
It is already assigned. So passing an empty string. But setting `cert.Subject.Organizations` will also work as we use `stringSetMerger{}` to merge later.

Reply:
Let's assign cert.Subject.Organizations . That is less confusing.

Reply:
updated